sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/ConversationAsyncClient.java

```diff
@@ -469,17 +486,199 @@
+    public Mono<Void> downloadTo(
+        OutputStream stream,
+        URI endpoint,
+        HttpRange range,
+        ParallelDownloadOptions parallelDownloadOptions) {
+
+        return Mono.defer(() -> {
+            downloadToWithResponse(stream, endpoint, range, parallelDownloadOptions).block();
+            return Mono.empty();
+        });
+    }
+
+    /**
+     * Download the recording content (e.g. Recording's metadata, Recording video, etc.) from the {@code endpoint}
+     * and written into the {@code stream}.
+     * @param stream - Stream to write the content to.
+     * @param endpoint - Location of the recording's content.
+     * @param range - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @param parallelDownloadOptions - An optional {@link ParallelDownloadOptions} object to modify how the
+     *                               parallel download will work.
+     * @return A {@link Mono} object containing the http response information from the download.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> downloadToWithResponse(
+        OutputStream stream,
+        URI endpoint,
+        HttpRange range,
+        ParallelDownloadOptions parallelDownloadOptions) {
+
+        return downloadToWithResponse(stream, endpoint, range, parallelDownloadOptions, null);
+    }
+
+    Mono<Response<Void>> downloadToWithResponse(
+        OutputStream stream,
+        URI endpoint,
+        HttpRange range,
+        ParallelDownloadOptions parallelDownloadOptions,
+        Context context) {
+        HttpRange finalRange = range == null ? new HttpRange(0) : range;
+        ParallelDownloadOptions finalParallelDownloadOptions =
+            parallelDownloadOptions == null
+                ? new ParallelDownloadOptions()
+                : parallelDownloadOptions;
+
+        return contentDownloader.downloadToStream(stream, endpoint, finalRange, finalParallelDownloadOptions, context);
+    }
+
+    /**
+     * Download the recording content (e.g. Recording's metadata, Recording video, etc.) from the {@code endpoint}
+     * and stored into the file in {@code path}.
+     * @param path - File path to store file to.
+     * @param endpoint - Location of the recording's content.
+     * @param range - An optional {@link HttpRange} value containing the range of bytes to download. If missing,
+     *                  the whole content will be downloaded.
+     * @param parallelDownloadOptions - An optional {@link ParallelDownloadOptions} object to modify how the
+     *                               parallel download will work.
+     * @param override - True to override the file if it exists.
+     * @return An empty {@link Mono} object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> downloadTo(
+        String path,
```

Comment:
May want to take `Path` instead of `String` in this API to more clearly indicate its requirement.